examples/web-wagmi/src/profiles/UseUpdateProfileImage.tsx

```diff
@@ -0,0 +1,156 @@
+import { ProfileFieldsFragment, useUpdateProfileImage } from '@lens-protocol/react';
+import { ChangeEvent, useState } from 'react';
+
+import { LoginButton } from '../components/auth/LoginButton';
+import { WhenLoggedIn, WhenLoggedOut } from '../components/auth/auth';
+import { useFilePreview } from '../hooks/useFilePreview';
+import { ILocalFile, ImageType, useFileSelect } from '../hooks/useFileSelect';
+import { uploadImage } from '../upload';
+import { SmallProfileCard } from './components/ProfileCard';
+
+function UpdateUploadedImage({ profile }: { profile: ProfileFieldsFragment }) {
+  const [inputValue, setInputValue] = useState(
+    'https://arweave.net/dOKOqiZVvSs14n54GIRH9nkSlLKArzK7-SPc2sBVmAM',
+  );
+  const { update, error, isPending } = useUpdateProfileImage({
+    profileId: profile.id,
+  });
+
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleUpdateProfileImage = async () => {
+    await update(inputValue);
+  };
+
+  return (
+    <div style={{ paddingTop: '2rem' }}>
+      <label>Image URL</label>
+      <input
+        onChange={handleInputChange}
+        value={inputValue}
+        style={{ width: '90%', margin: '0' }}
+      />
+      <button onClick={handleUpdateProfileImage} disabled={isPending}>
+        Update profile image
+      </button>
+      {error && <p>{error.message}</p>}
+      {isPending && <p>Updating your Lens profile's image...</p>}
+    </div>
+  );
+}
+
+function UploadAndUpdateNewImage({ profile }: { profile: ProfileFieldsFragment }) {
+  const [candidateFile, setCandidateFile] = useState<ILocalFile<ImageType> | null>(null);
+  const [uploadError, setUploadError] = useState<string>();
```

Comment:
nit: imo it's better to have this typed as the the Error class that can be thrown or null, so we can better distinguish between error or no error. Or maybe even just `string | null`

then we can avoid setting to an empty string to signal there is no error